Zadanie1/Zadanie2

Kmynek opened the pull request.

Kmynek committed:
Zadanie1-adapter

Kmynek committed:
Zadanie2-command

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DriverAdapterForFiguresJane.java:
```diff
@@ -0,0 +1,35 @@
+//package edu.kis.powp.jobs2d.drivers.adapter;
+//
+//import edu.kis.powp.jobs2d.AbstractDriver;
+//import edu.kis.powp.jobs2d.Job2dDriver;
+//import edu.kis.powp.jobs2d.features.DriverFeature;
+//
+//public class DriverAdapterForFiguresJane extends AbstractDriver {
```
What is that comment for?

kowallus commented on src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java:
```diff
@@ -83,7 +98,7 @@ public static void main(String[] args) {
 				Application app = new Application("2d jobs Visio");
 				DrawerFeature.setupDrawerPlugin(app);
-				setupDefaultDrawerVisibilityManagement(app);
+				//setupDefaultDrawerVisibilityManagement(app);
```
In revision repo it is safe to delete code instead of using a comment.

kowallus commented on src/test/java/edu/kis/powp/jobs2d/TestJobs2dPatterns.java:
```diff
@@ -26,7 +30,14 @@ private static void setupPresetTests(Application application) {
+		application.addTest("Figure Joe 2", (ActionEvent e) -> {
+            FiguresJoe.figureScript2(DriverFeature.getDriverManager().getCurrentDriver());
+        });
+//		application.addTest("Figures Jane", (ActionEvent e) -> {
+//		    FiguresJane.figureScript(new DriverAdapterForFiguresJane());
+//		});
```
Do not commit code that does not work

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/MyAdapter.java:
```diff
@@ -4,15 +4,16 @@
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class MyAdapter implements Job2dDriver {
```
You should fix the name of this class: "nieprecyzyjna nazwa klasy"

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/MyAdapter.java:
```diff
@@ -4,15 +4,16 @@
+	private DrawPanelController drawer;
 	public MyAdapter() {
-		super();
+		this.drawer= DrawerFeature.getDrawerController();
```
Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/DriverAdapterForFiguresJane.java:
```diff
@@ -0,0 +1,35 @@
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DriverFeature;
+
+public class DriverAdapterForFiguresJane implements Job2dDriver {
```
This class does not solve the problem of FiguresJane patterns.

kowallus replied:
This is not obligatorary. I assume it still does not work. There is no test added with FiguresJane script.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    public void setLineThickness(int thickness) {
+        this.lineThickness = thickness;
```
This does not seem to work.

kowallus commented on src/main/java/edu/kis/powp/command/OperateToCommand.java:
```diff
@@ -0,0 +1,20 @@
+public class OperateToCommand implements DriverCommand {
+    private int x;
+    private int y;
+    private Job2dDriver driver;
```
The solution lacks flexibility, using different drivers with commands is difficult (e.g., you could consider setter).

kowallus commented on src/main/java/edu/kis/powp/command/factory/RectangleCommandFactory.java:
```diff
@@ -0,0 +1,19 @@
+import edu.kis.powp.jobs2d.Job2dDriver;
+
+public class RectangleCommandFactory {
+    public static DriverCommand createRectangleCommand(Job2dDriver driver, int x, int y, int width, int height) {
```
"Zaimplementuj kilka „fabryk”" I see only one

Kmynek committed:
Fix Z1/Z2

-deleted unused code
-changed name of MyAdapter to Job2dAdapter
-fixed the constructor injection for Job2dAdapter
-Fixed line thickness - now we can simulate changing line parameters in program.
-Setter added to OperateToCommand
-added two more fabrics for circle and triangle.

kowallus commented on .idea/compiler.xml:
Commits polluted with project files (.settings, .idea, etc.).

kowallus commented on src/main/java/edu/kis/powp/command/SetPositionCommand.java:
```diff
@@ -0,0 +1,20 @@
+public class SetPositionCommand implements DriverCommand {
+    private int x;
+    private int y;
+    private Job2dDriver driver;
```
Consequently, setters should be applied to other classes in DriverCommand hierarchy.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dDrawerAdapter.java:
```diff
@@ -4,15 +4,17 @@
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class Job2dDrawerAdapter implements Job2dDriver {
```
The name is better but it does not indicate the purpose of the adapter or what was actually adapted.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/Job2dDrawerAdapter.java:
```diff
@@ -4,15 +4,17 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
```
unused import

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,100 @@
+
+    public LineDrawerAdapter(LineMode mode) {
+        this.mode = mode;
+        this.drawer = DrawerFeature.getDrawerController();
```
unnecessary dependency

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ThickLineDecorator.java:
```diff
@@ -0,0 +1,73 @@
+    @Override
+    public float getThickness() {
+        // returns thickness
+        return thickness *10.0f;
```
this doesn't look necessary.

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/ThickLineDecorator.java:
This class is not used. Does it work?

kowallus commented on src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:
```diff
@@ -0,0 +1,100 @@
+    }
+
+    private ILine createLine(int x1, int y1, int x2, int y2) {
+        ILine line;
+        if(mode == LineMode.SPECIAL) {
+            line = LineFactory.getSpecialLine();
+        } else {
+            line = LineFactory.getBasicLine();
```
Unfortunately, this is not a generic solution. (e.g. new line type requires modifications in this class, unnecessary direct coupling with LineFactory). It is better simply to inject ILine dependency (then you can test ThickLineDecorator).

kowallus reviewed (changes requested):
Not quite yet.

Kmynek committed:
Fix Z1/Z2 vol.2

Refactor project:
- Update .gitignore to exclude IDE and build files.
- Add setters to commands and update shape factories.
- Rename Job2dDrawerAdapter to DrawPanelToJob2dDriverAdapter.
- Inject dependencies in LineDrawerAdapter and use method references.
- Remove unused classes.

kowallus commented on src/main/java/edu/kis/powp/command/ComplexCommand.java:
```diff
@@ -0,0 +1,19 @@
+
+public class ComplexCommand implements DriverCommand {
+    private List<DriverCommand> commands = new ArrayList<>();
+
```
There should be also a setter for driver used in child commands.

kowallus commented on src/main/java/edu/kis/powp/command/OperateToCommand.java:
```diff
@@ -0,0 +1,33 @@
+        this.driver = driver;
+    }
+
+    public void setX(int x) {
+        this.x = x;
+    }
+
+    public void setY(int y) {
+        this.y = y;
+    }
```
OK, but seems unnecessary, too fine-grained.

kowallus commented on src/main/java/edu/kis/powp/command/factory/TriangleCommandFactory.java:
```diff
@@ -0,0 +1,34 @@
+        ComplexCommand triangleCommand = new ComplexCommand();
+        //start point - left corner
+        SetPositionCommand setPosCommand = new SetPositionCommand(driver, 0, 0);
+        setPosCommand.setX(x);
+        setPosCommand.setY(y);
```
I cannot see the point of such a way of setting X and Y.

kowallus reviewed (approved):
OK
